### ISSUE TITLE
Error: `could not find application file: erlang_v8.app`

### ISSUE THREAD

**Issue by piglovesyou** [le0pard/elixir_v8#5]
In my personal project, after I put `{:elixir_v8, "~> 0.2.2"}` as deps in mix.exs, hit `$ mix deps.get` and `$ iex -S mix`, I see this:

```
Erlang/OTP 19 [erts-8.2] [source] [64-bit] [smp:8:8] [async-threads:10] [hipe] [kernel-poll:false]

Compiling 1 file (.ex)
Generated elixir_v8_try app

=INFO REPORT==== 16-Feb-2017::17:21:05 ===
    application: logger
    exited: stopped
    type: temporary
** (Mix) Could not start application erlang_v8: could not find application file: erlang_v8.app
```

Did I miss something?

**Comment by le0pard** [comment]
Maybe `mix deps.compile --all`?

**Comment by piglovesyou** [comment]
Hmm still get `{:error, {'no such file or directory', 'Elixir.ElixirV8.app'}}` even in a cloned `elixir_v8` directory.
```
[pig@localhost elixir_v8]$ mix deps.compile --all
===> Compiling erlang_v8
make[1]: Entering directory '/home/pig/git/elixir_v8/deps/erlang_v8/deps/jsx'
make[1]: Leaving directory '/home/pig/git/elixir_v8/deps/erlang_v8/deps/jsx'
make -C /home/pig/git/elixir_v8/deps/erlang_v8/c_src
make[1]: Entering directory '/home/pig/git/elixir_v8/deps/erlang_v8/c_src'
make[1]: Nothing to be done for 'all'.
make[1]: Leaving directory '/home/pig/git/elixir_v8/deps/erlang_v8/c_src'
==> poolboy (compile)

(...some warnings of "please use parentheses")

[pig@localhost elixir_v8]$ iex -S mix
Erlang/OTP 19 [erts-8.2] [source] [64-bit] [smp:8:8] [async-threads:10] [hipe] [kernel-poll:false]

Compiling 2 files (.ex)

(...some warnings of "please use parentheses")

Generated elixir_v8 app
Interactive Elixir (1.4.1) - press Ctrl+C to exit (type h() ENTER for help)
iex(1)> ElixirV8.start
{:error, {'no such file or directory', 'Elixir.ElixirV8.app'}}
```
A build of `erlang_v8` seems succeeded.
```
(...)
[943/945] SOLINK ./libv8.so
[944/945] CXX obj/d8/d8.o
[945/945] LINK ./d8
g++ \
        -I/home/pig/git/elixir_v8/deps/erlang_v8/c_src/../build/v8/include \
        erlang_v8.cc report.cc vm.cc \
        -o /home/pig/git/elixir_v8/deps/erlang_v8/c_src/../priv/erlang_v8 \
        -Wl,/home/pig/git/elixir_v8/deps/erlang_v8/c_src/../priv/libv8{,_libbase,_libplatform}.so \
        -Wl,/home/pig/git/elixir_v8/deps/erlang_v8/c_src/../priv/libicu{uc,i18n}.so \
        -Wl,-rpath,'$ORIGIN'/. \
        -lrt \
        -ldl \
        -pthread \
        -std=c++11
make[1]: Leaving directory '/home/pig/git/elixir_v8/deps/erlang_v8/c_src'
Compiling 2 files (.ex)

(...some warnings of "please use parentheses")

Generated elixir_v8 app
Interactive Elixir (1.4.1) - press Ctrl+C to exit (type h() ENTER for help)
```
I think I need to learn elixir's `application` and where the `*.app` file should be located.

**Comment by le0pard** [comment]
You no need `ElixirV8.start` if you add application to section `application` in mix.

**Comment by piglovesyou** [comment]
Thanks, but I'm in a cloned `elixir_v8` directory now

**Comment by le0pard** [comment]
I do not use elixir_v8 as mix app, I use it with inside another apps, so don't know, what do you want to do with it.

**Comment by piglovesyou** [comment]
I'd like to use elixir_v8 in my Phoenix app actually to make response html by React.js server side rendering. I needed to make sure how elixir_v8 works at least so I tried `iex -S mix` in elixir_v8 directory, but still cannot run `exec` function successfully...

**Comment by le0pard** [comment]
https://github.com/le0pard/js-api-service - it is working inside Phoenix app (at least 1.5 year ago)

**Comment by piglovesyou** [comment]
Thank you, let me try that.

**Comment by piglovesyou** [comment]
I got this: `Compilation error on file lib/phoenix/code_reloader.ex`. Maybe this is just a deps version problem. I'm trying to fix that.
https://gist.github.com/piglovesyou/8c0b7e3a0efbef85769f8023425c8a41

**Comment by piglovesyou** [comment]
After fixing deps versions it launches, but `POST /run` returns the following. Seems that more fix is needed.

```
[pig@localhost js-api-service]$ mix phoenix.server
[info] Running JsApiService.Endpoint with Cowboy using http://localhost:4000
17 Feb 00:29:34 - info: compiled 3 files into 2 files in 718ms
[info] POST /run
[debug] Processing by JsApiService.JsFunctionController.run/2
  Parameters: %{}
  Pipelines: []
[info] Sent 200 in 38ms
[error] #PID<0.408.0> running JsApiService.Endpoint terminated
Server: localhost:4000 (http)
Request: POST /run
** (exit) an exception was raised:
    ** (Plug.Conn.AlreadySentError) the response was already sent
        (plug) lib/plug/conn.ex:496: Plug.Conn.resp/3
        (plug) lib/plug/conn.ex:483: Plug.Conn.send_resp/3
        (js_api_service) web/controllers/js_function_controller.ex:1: JsApiService.JsFunctionController.action/2
        (js_api_service) web/controllers/js_function_controller.ex:1: JsApiService.JsFunctionController.phoenix_controller_pipeline/2
        (js_api_service) lib/js_api_service/endpoint.ex:1: JsApiService.Endpoint.instrument/4
        (js_api_service) lib/phoenix/router.ex:261: JsApiService.Router.dispatch/2
        (js_api_service) web/router.ex:1: JsApiService.Router.do_call/2
        (js_api_service) lib/js_api_service/endpoint.ex:1: JsApiService.Endpoint.phoenix_pipeline/1
```

**Comment by piglovesyou** [comment]
http://stackoverflow.com/a/35250468/804314 solved the above problem. This is another error, maybe I should post it as an erlang_v8 issue?

```
[info] POST /run
[debug] Processing by JsApiService.JsFunctionController.run/2
  Parameters: %{"data" => "2", "function" => "1+1"}
  Pipelines: []
[error] GenServer #PID<0.324.0> terminating
** (ArgumentError) argument error
    (erlang_v8) /home/pig/git/js-api-service/deps/erlang_v8/src/erlang_v8_vm.erl:239: :erlang_v8_vm.send_to_port/5
    (erlang_v8) /home/pig/git/js-api-service/deps/erlang_v8/src/erlang_v8_vm.erl:104: :erlang_v8_vm.handle_call/3
    (stdlib) gen_server.erl:615: :gen_server.try_handle_call/4
    (stdlib) gen_server.erl:647: :gen_server.handle_msg/5
    (stdlib) proc_lib.erl:247: :proc_lib.init_p_do_apply/3
Last message: {:eval, "(function(){ var data = arguments[0]; 1+1}).apply(null, [\"2\"]);", 5000, 2000}
State: {:state, "//     Underscore.js 1.8.3\n// (...)
[info] Sent 500 in 67ms
[error] #PID<0.464.0> running JsApiService.Endpoint terminated
Server: localhost:4000 (http)
Request: POST /run
** (exit) exited in: :gen_server.call(#PID<0.324.0>, {:eval, "(function(){ var data = arguments[0]; 1+1}).apply(null, [\"2\"]);", 5000, 2000}, 30000)
    ** (EXIT) an exception was raised:
        ** (ArgumentError) argument error
            (erlang_v8) /home/pig/git/js-api-service/deps/erlang_v8/src/erlang_v8_vm.erl:239: :erlang_v8_vm.send_to_port/5
            (erlang_v8) /home/pig/git/js-api-service/deps/erlang_v8/src/erlang_v8_vm.erl:104: :erlang_v8_vm.handle_call/3
            (stdlib) gen_server.erl:615: :gen_server.try_handle_call/4
            (stdlib) gen_server.erl:647: :gen_server.handle_msg/5
            (stdlib) proc_lib.erl:247: :proc_lib.init_p_do_apply/3
```

**Comment by le0pard** [comment]
it should work, because `"erlang_v8": {:git, "https://github.com/strange/erlang-v8.git", "1ca13321d5746f3ec20f0c0bfd2bcc8293db71da", []}` locked to commit.

**Comment by piglovesyou** [comment]
Here is my assumption. In short, we need to update elixir_v8 because of 404 error of `http://gyp.googlecode.com/svn/trunk`.

The current elixir_v8 fails building with erlang_v8 build failure log:

```
svn checkout --force http://gyp.googlecode.com/svn/trunk build/gyp \
    --revision 1831
svn: E170013: Unable to connect to a repository at URL 'http://gyp.googlecode.com/svn/trunk'
svn: E160013: '/svn/trunk' path not found
make[1]: *** [Makefile:443: dependencies] Error 1
```

So I updated elixir_v8 despite your `mix.lock`. Also I got another problem of building erlang_v8 so [I asked and had them fix](https://github.com/strange/erlang_v8/issues/4).

Now I got old elixir_v8 and new erlang_v8. The API of erlang_v8 that elixir_v8 expects is changed, so we got `** (ArgumentError) argument error`.

Let me check further.

Edit: They fixed it https://github.com/strange/erlang_v8/issues/3

**Comment by piglovesyou** [comment]
All right, the API was totally changed at least in `erlang_v8_vm.erl`.

Before: `eval(Pid, Source, Timeout) ->`
After: `eval(Pid, Context, Source) ->`

**Comment by devmarco** [comment]
Hey, Have you guys released a new version with this change? I am facing the same issue using `0.2.2`

Thanks btw. Good job

**Comment by le0pard** [comment]
@devmarco nope, not released. You can create separate issue for this - I will resolve this .

**Comment by devmarco** [comment]
@le0pard Done, https://github.com/le0pard/elixir_v8/issues/8

**Comment by yjukaku** [comment]
So I'm still getting this same error message with elixir_v8 version 0.3.0 . Thanks for all the work you guys have done to fix it, but is there any reason you can think of that this is still happening?

When I run `mix deps.compile --all` from my personal app:
```
...
==> elixir_v8
Compiling 3 files (.ex)
warning: function :erlang_v8.eval/4 is undefined (module :erlang_v8 is not available)
  lib/elixir_v8/vm.ex:13

warning: function :erlang_v8.start_vm/1 is undefined (module :erlang_v8 is not available)
  lib/elixir_v8/vm.ex:18

warning: function :erlang_v8.create_context/1 is undefined (module :erlang_v8 is not available)
  lib/elixir_v8/vm.ex:19

Generated elixir_v8 app
...
```

Then when I try to run `iex -S mix` with `:elixir_v8` in the mix.exs `applications` function:

```
** (Mix) Could not start application erlang_v8: could not find application file: erlang_v8.app
```

**Comment by le0pard** [comment]
@yjukaku you should use version directly from git, because mix is not allow publish package with git dependency.
